tableauserverclient/helpers/strings.py

```diff
@@ -0,0 +1,65 @@
+import requests
+
+from defusedxml.ElementTree import fromstring, tostring
+from functools import singledispatch
+from typing import TypeVar
+
+
+# the redact method can handle either strings or bytes, but it can't mix them.
+# Generic type so we can write the actual logic once, then use singledispatch to
+# create the replacement text with the correct type
+T = TypeVar("T", str, bytes)
+
+
+# TODO: ideally this would be in the logging config
+def safe_to_log(server_response: requests.Response) -> str:
+    """Checks if the server_response content is not xml (eg binary image or zip)
+    and replaces it with a constant"""
+    ALLOWED_CONTENT_TYPES = ("application/xml", "application/xml;charset=utf-8")
+    if server_response.headers.get("Content-Type", None) not in ALLOWED_CONTENT_TYPES:
+        return "[Truncated File Contents]"
```

Comment:
Is "Redacted" more correct here?

Reply:
just a thought

Reply:
Hmm, I think better not to say "redacted" because it isn't using our "redaction" method, but yea, truncated sounds inaccurate. Not sure what would be a good alternative.